cli/popper/commands/cmd_dot.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+import click
+import hcl
+import popper.utils as pu
+from ..cli import pass_context
+from popper.gha import Workflow
+
+
+
+@click.option(
+    '--wfile',
+    help=(
+        'File containing the definition of the workflow. '
+        '[default: ./github/main.workflow OR ./main.workflow]'
+    ),
+    required=False,
+    default=None
+)
+@click.option(
+    '--recursive',
+    help='Generate .dot file for any.workflow file '
+         'found recursively from current path.',
+    required=False,
+    is_flag=True
+)
+@click.command('dot', short_help='Generates a .dot format for a given workflow'
+               '\n[.dot is a format for graphical representation]')
+@pass_context
+def cli(ctx, wfile, recursive):
+    """
+    Analyzes a workflow file and then creates a dot representation for the same
+    .dot represents the order in which actions in the workflow must be executed
+    """
+    wfile_list = list()
+    if recursive:
+        wfile_list = pu.find_recursive_wfile()
+    else:
+        wfile_list.append(pu.find_default_wfile(wfile))
+
+    for wfile in wfile_list:
+        pipeline = Workflow(wfile, False, False, False, False)
+
+        wf = pipeline.wf
+        name = list(wf['workflow'].keys())[0]
+
+        action = wf['resolves'][0]
+        parent_action = cur_action = action
+
+        graph = list()
+        graph = add(parent_action, cur_action, wf['action'], graph)
+        graph = ''.join(list(set(graph)))
+        graph = "digraph G {\n" + graph + "}\n"
+        pu.info(graph)
+
+
+# Recursively go through "needs" and add corresponding actions to graph
```

Comment:
seems like the logic is inverted and this ends up generating an inverted representation of the graph.

One alternative is to use the `next` attribute of each node, which represent forward-going edges ([this block of code](https://github.com/systemslab/popper/blob/master/cli/popper/gha.py#L168-L171) adds these). So you can start from the root and then use `next` to get to the subsequent nodes.

Reply:
Yes, I had to invert the order to get the output below by using `popper dot | graph-easy`. This is I think because of how directed graphs are. Should I change it to an  undirected graph?

> Resolve issure #513. `popper view` now displays ascii-art diag for a workflow.
> 
> Example: This [workflow](https://github.com/cplee/github-actions-demo/blob/master/.github/main.workflow) is represented as
> 
> ```
>              +---------------+
>              |    deploy     |
>              +---------------+
>                |
>                |
>                v
> +------+     +---------------+
> | test | <-- | branch_filter |
> +------+     +---------------+
>   |            |
>   |            |
>   |            v
>   |          +---------------+
>   |          |     lint      |
>   |          +---------------+
>   |            |
>   |            |
>   |            v
>   |          +---------------+
>   +--------> |    install    |
>              +---------------+
> ```

Reply:
in the diagram you linked to, the order of actions is inverted. The graph should look like this:

```
install -> test -> branch-filter -> deploy
           -> lint 
```

the first action that executes is the install action, followed by test and lint, then branch-filter and then deploy. In the diagram you posted above, deploy is the first action that gets executed

Reply:
Oh.. I will change it to reflect that